### DISCUSSION TITLE
Walrus Operator for C# variable assigning

### DISCUSSION THREAD

**Discussion by GitisAwesome** [dotnet/csharplang#7423] [Language Ideas] [answered]
Maybe add in a walrus operator that guesses what type of variable you are declaring like in Go. It could help make declaring variables faster but the variable's value must be written down fully like for integers: a := 5;, for floats, decimals, or doubles: a := 5.0;, for characters: a := "a";, for strings:  a := "Hello World";. 

Thank you for reading this suggestion.

**Comment by HaloFour** [comment, marked as answer]
C# already has type inference for local variables through the `var` keyword:

```cs
var s = "Hello World";
var d = 5.0;
var f = 5.0f;
var i = 5;
```

**Reply by GitisAwesome** [reply]
Thank you for responding, I see what you mean I learnt where you had to put the type before but this is works. At least I learnt something new. Thank you.
